--- baseline.c
+++ candidates/M15.c
@@ -34,17 +34,17 @@
     }
     if(right<x || bottom<y)return;
     if(!D_8014A248) {
-        if((D_8012E608&4)&&(D_8012E608&8)) {
-            texture_edge=s+right;s=texture_edge-x;t+=bottom-y;
-            gSPTextureRectangle(D_80149438++,x<<2,y<<2,right<<2,bottom<<2,0,s<<5,t<<5,-4096,-1024);
-        } else if(D_8012E608&4) {
+        {
+            int command_dx,command_t,command_dy;
+            if(D_8012E608&4) {
             texture_edge=s+right;s=texture_edge-x;
-            gSPTextureRectangle(D_80149438++,x<<2,y<<2,right<<2,bottom<<2,0,s<<5,t<<5,-4096,1024);
-        } else if(D_8012E608&8) {
+                command_dx=-4096;
+            } else {command_dx=4096;}
+            if(D_8012E608&8) {
             t+=bottom-y;
-            gSPTextureRectangle(D_80149438++,x<<2,y<<2,right<<2,bottom<<2,0,s<<5,t<<5,4096,-1024);
-        } else {
-            gSPTextureRectangle(D_80149438++,x<<2,y<<2,right<<2,bottom<<2,0,s<<5,t<<5,4096,1024);
+                command_t=t<<5;command_dy=-1024;
+            } else {command_t=t<<5;command_dy=1024;}
+            gSPTextureRectangle(D_80149438++,x<<2,y<<2,right<<2,bottom<<2,0,s<<5,command_t,command_dx,command_dy);
         }
     } else {
         height=bottom-y;
